### PR TITLE
doc: clarify the contract of get(String)

### PR DESCRIPTION
**Pull request by monperrus**
Spin off from #2982

### CODE REVIEW

File: src/main/java/spoon/reflect/factory/TypeFactory.java [hunk 1]
```diff
@@ -402,9 +402,13 @@ public CtTypeParameterReference createReference(CtTypeParameter type) {
...
 
 	/**
-	 * Gets a created type from its qualified name.
+	 * Gets a created type from its qualified name if source in the source classpath. 
 	 *
-	 * @return a found type or null if does not exist
+	 * `TypeFactory#get(String)` returns null if the class is not in the source classpath (even if it is in the binary classpath).
+	 * `TypeFactory#get(Class)` returns null if the class is neither in the source classpath nor in the binary classpath, 
+	 * and returns a [shadow class](http://spoon.gforge.inria.fr/reflection.html) if it is only in the binary classpath. 
+	 * Note that a shadow class has an empty method body, if you need a shadow class with method body, see [spoon-decompiler](https://github.com/INRIA/spoon/tree/master/spoon-decompiler))
```
Comment (nharrand) [inline src/main/java/spoon/reflect/factory/TypeFactory.java]:
```suggestion
	 * Note that a shadow class has empty method bodies, if you need a shadow class with method bodies, see [spoon-decompiler](https://github.com/INRIA/spoon/tree/master/spoon-decompiler))
```

File: src/main/java/spoon/reflect/factory/TypeFactory.java [hunk 2]
```diff
@@ -402,9 +402,13 @@ public CtTypeParameterReference createReference(CtTypeParameter type) {
 	}
 
 	/**
-	 * Gets a created type from its qualified name.
+	 * Gets a created type from its qualified name if source in the source classpath.
```
Comment (nharrand) [inline src/main/java/spoon/reflect/factory/TypeFactory.java]:
```suggestion
	 * Gets a created type from its qualified name if source in the source classpath.
```

File: src/main/java/spoon/reflect/factory/TypeFactory.java [hunk 3]
```diff
@@ -402,9 +402,13 @@ public CtTypeParameterReference createReference(CtTypeParameter type) {
 	}
 
 	/**
-	 * Gets a created type from its qualified name.
+	 * Gets a created type from its qualified name if source in the source classpath. 
 	 *
-	 * @return a found type or null if does not exist
+	 * `TypeFactory#get(String)` returns null if the class is not in the source classpath (even if it is in the binary classpath).
+	 * `TypeFactory#get(Class)` returns null if the class is neither in the source classpath nor in the binary classpath,
```
Comment (nharrand) [inline src/main/java/spoon/reflect/factory/TypeFactory.java]:
```suggestion
	 * `TypeFactory#get(Class)` returns null if the class is neither in the source classpath nor in the binary classpath,
```

File: src/main/java/spoon/reflect/factory/TypeFactory.java [hunk 4]
```diff
@@ -402,9 +402,13 @@ public CtTypeParameterReference createReference(CtTypeParameter type) {
 	}
 
 	/**
-	 * Gets a created type from its qualified name.
+	 * Gets a created type from its qualified name if source in the source classpath. 
 	 *
-	 * @return a found type or null if does not exist
+	 * `TypeFactory#get(String)` returns null if the class is not in the source classpath (even if it is in the binary classpath).
+	 * `TypeFactory#get(Class)` returns null if the class is neither in the source classpath nor in the binary classpath, 
+	 * and returns a [shadow class](http://spoon.gforge.inria.fr/reflection.html) if it is only in the binary classpath.
```
Comment (nharrand) [inline src/main/java/spoon/reflect/factory/TypeFactory.java]:
```suggestion
	 * and returns a [shadow class](http://spoon.gforge.inria.fr/reflection.html) if it is only in the binary classpath.
```

**Comment by monperrus** [comment]
Thanks @nharrand, I love the commit suggestion feature, it's so productive! Don't hesitate to modify the PR directly also :-)

**Pull request merged**